plugins/microchip/switchtec-nvme-device.c

```diff
@@ -100,6 +113,61 @@ int pax_nvme_submit_admin_passthru(int fd, struct nvme_passthru_cmd *cmd)
 	return status;
 }
 
+int pax_nvme_submit_admin_passthru(int fd, struct nvme_passthru_cmd *cmd)
+{
+	int ret;
+
+	struct pax_nvme_device *pax;
+	struct switchtec_adm_passthru_nvme_req req;
+	struct switchtec_adm_passthru_nvme_rsp rsp;
+	size_t data_len;
+	size_t rsp_len;
+	int status;
+
+	memset(&req, 0, sizeof(req));
+	memset(&rsp, 0, sizeof(rsp));
+	pax = to_pax_nvme_device(global_device);
+
+	memcpy(&req.nvme_sqe, cmd, 16 * 4);
+
+	/* sqe[9] should be 0 per spec */
+	req.nvme_sqe[9] = 0;
+
+	if (cmd->data_len > sizeof(req.nvme_data))
+		data_len = sizeof(req.nvme_data);
+	else
+		data_len = cmd->data_len;
+
+	memcpy(req.nvme_data, (void *)cmd->addr, data_len);
+	data_len = data_len + 16 * 4;
+
+	rsp_len = (sizeof(rsp.nvme_cqe) + cmd->data_len);
+
+	ret = switchtec_nvme_admin_passthru(pax->dev, pax->pdfid,
+					    data_len, &req,
+					    &rsp_len, &rsp);
+	if (ret && errno == EBADMSG) {
```

Comment:
EBADMSG from firmware? Should be UNSUPPORTED?

Reply:
Yes this is a problem. Let's discuss.